src/analysis.ts

```diff
@@ -1,12 +1,147 @@
 // This file will hold all the functions that do analysis of packages.
+import * as acorn from 'acorn';
+import {CallExpression, Node} from 'estree';
+import {PointOfInterest, Position} from './package-tree';
 
+const walk = require('acorn/dist/walk');
 
+export const ioModuleList: string[] =
+    ['http', 'fs', 'https', 'http2', 'net', 'datagram', 'child_process'];
+
+/**
+ * Gets a list of PointOfInterest objects, indicating that there were IO
+ * modules that were required in a file
+ *
+ * @param contents the contents of the file
+ * @param file the name of the file being checked
+ */
 export function getIOModules(
-    contents: string, fileName: string): PointOfInterest[] {
-  throw new Error('not implemented');
+    contents: string, file: string): PointOfInterest[] {
+  const acornTree = acorn.parse(contents);
+  const requireCalls = getRequireCalls(acornTree);
+  const moduleList = getRequiredModules(requireCalls, file);
+  const requiredIOModules =
+      moduleList.filter(mod => ioModuleList.includes(mod.type));
+  return requiredIOModules;
 }
 
+/**
+ * Gets a list of PointOfInterest objects, indicating that dynamic
+ * evaluation was used in the file
+ *
+ * @param contents the contents of the file
+ * @param file the name of the file being checked
+ */
 export function getDynamicEval(
-    contents: string, fileName: string): PointOfInterest[] {
-  throw new Error('not implemented');
-}
\ No newline at end of file
+    contents: string, file: string): PointOfInterest[] {
+  const acornTree = acorn.parse(contents);
+  const dynamicEval = getDynamicRequireCalls(acornTree, file);
+  return dynamicEval;
+}
+
+/**
+ * Returns a list of acorn nodes that contain 'require' call expressions
+ *
+ * @param tree abstract syntax tree
+ */
+function getRequireCalls(tree: Node) {
+  const requireCalls: Node[] = [];
+  walk.simple(tree, {
+    CallExpression(e: CallExpression) {
+      if (e.callee.type === 'Identifier' && e.callee.name === 'require') {
+        requireCalls.push(e.arguments[0]);
+      }
+    }
+  });
+  return requireCalls;
+}
+
+/**
+ * Iterates through nodes with the name 'require' and returns a list of
+ * PointOfInterest objects that indicate specific io modules that were required
+ *
+ * @param requireNodes array of acorn nodes that contain the 'require' call
+ *    expression
+ * @param file the name of the file being checked
+ */
+function getRequiredModules(
+    requireNodes: Node[], file: string): PointOfInterest[] {
+  const requiredModules: PointOfInterest[] = [];
+  const dynamicArgPos: Position[] = [];
+  requireNodes.forEach((node: Node) => {
```

Comment:
Super minor nit: I generally stay away from `forEach` with a function literal. `for...of` loops serve the same purpose and are easier to read and easier to debug.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of